### ISSUE TITLE
GCE internal load balancer always in pending state.

### ISSUE THREAD

**Issue by burnyd** [kubernetes/kops#15176]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

kops version
I0222 10:44:55.346029    4238 featureflag.go:163] Unknown FeatureFlag "AlphaAllowGCE"
Client version: 1.25.3 (git-v1.25.3)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

 kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.0", GitCommit:"b46a3f887ca979b1a5d14fd39cb1af43e7e5d12d", GitTreeState:"clean", BuildDate:"2022-12-08T19:58:30Z", GoVersion:"go1.19.4", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.4", GitCommit:"872a965c6c6526caa949f0c6ac028ef7aff3fb78", GitTreeState:"clean", BuildDate:"2022-11-09T13:29:58Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"linux/amd64"}

**3. What cloud provider are you using?**

GCE

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

When adding either annotation of **cloud.google.com/load-balancer-type: "internal"** or **networking.gke.io/load-balancer-type "Internal"**

The service is always stuck in a <pending> state.

**6. What did you expect to happen?**

Expectation is that the service would get a private ip address for type: Loadbalancer as opposed to a public IP address.

This was the logging for it..

I0221 18:42:01.630638       1 controller.go:403] Ensuring load balancer for service ceosnms/vxlan-service-internal-test
I0221 18:42:01.630680       1 controller.go:912] Adding finalizer to service ceosnms/vxlan-service-internal-test
I0221 18:42:01.631378       1 event.go:294] "Event occurred" object="ceosnms/vxlan-service-internal-test" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
I0221 18:42:01.713718       1 gce_loadbalancer_internal.go:81] ensureInternalLoadBalancer(ad1644878e1304ad994558afb13822e7): Attaching "[gke.networking.io/l4-ilb-v1](http://gke.networking.io/l4-ilb-v1)" finalizer
E0221 18:42:01.715499       1 gce_loadbalancer_internal.go:83] Failed to attach finalizer '[gke.networking.io/l4-ilb-v1](http://gke.networking.io/l4-ilb-v1)' on service ceosnms/vxlan-service-internal-test - services "vxlan-service-internal-test" is forbidden: **User "system:serviceaccount:kube-system:cloud-provider" cannot patch resource "services/status" in API group** "" in the namespace "ceosnms"

So I had to create a new cluster binding and role for that account with the following.
```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: gce:cloud-provider
  namespace: ceosnms
rules:
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
  - patch
  - update
- apiGroups:
  - ""
  resources:
  - services/status
  verbs:
  - patch
  - update
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  namespace: ceosnms
  name: gce:cloud-provider
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: gce:cloud-provider
subjects:
- kind: ServiceAccount
  name: cloud-provider
  namespace: kube-system
```

I am wondering if these permissions can be added for GCE deployments for the controller?  Just doing the regular type load balancer without any sort of options just works but apparently any annotations that cause the cloud controller to do differently it needs permissions.

**Comment by johngmyers** [comment]
Office hours: cherrypick to 1.27

**Comment by hakman** [comment]
This should be fixed for 1.27.0. Thanks for reporting @burnyd!
